### ISSUE TITLE
building tensorflow from source / general issue

### ISSUE THREAD

**Issue by srvoin** [tensorflow/tensorflow#12646]
Dear Tensorflow team,

I was facing a problem described here:
https://stackoverflow.com/questions/45859371/error-building-tensorflow-on-centos-7

In short words: In the documentation there is no any single word about bazel version need for specific version of Tensorflow. In particular, I could not build a Python package with the newest version of Bazel. Experimentally I found out that that the master version (r1.3) can be built successfully with Bazel not newer than 0.5.1. (with very old versions it also doesn't work). 
So I am kindly asking to mention a Bazel version (as well as other dependencies, such as gcc, etc) to have successful build for the next releases.

I spent a lot of time to find a solution. Probably it will be useful experience for others.

Cheers,
Sergey

**Comment by martinwicke** [comment]
gunan@ is our bazel version check not working again? 

@srvoin If the check is working properly, it should tell you something like "you need at least bazel X.X.X".

I think otherwise this is an unfortunate one-off, since the 0.5.3 bazel release was broken.

Leaving open only to check whether our check works.

**Comment by gunan** [comment]
@yifeif are you aware of any issues with bazel version check?

**Comment by yifeif** [comment]
configure checks bazel against a minimum version of 0.4.5.

afaik Bazel 0.5.3 issue was resolved with #11871. I haven't seen the swig error mentioned in the SO link.

**Comment by srvoin** [comment]
@martinwicke as @yifeif said it were checking against a minimum version and nothing about maximum. In my case it didn't work with any version higher than 0.5.1
It looks like there are some differences in environment settings within bazel scope... could be this a problem: 
```
Release 0.5.3
Incompatible changes:

deprecate --python2_path and --python3_path flags for the
new --python_path flag; introduce --python_top flag for using
the new py_runtime rule.
....
```
I can't reproduce test with version 0.5.2 at the moment, but 0.5.3 and 0.5.4 don't work for python package generation.

**Comment by martinwicke** [comment]
Maybe we should fix the bazel version for releases, since it's breaking relatively quickly. 

This should be fixed at head though. @srvoin can you verify that building head with the latest bazel works for you?

**Comment by yifeif** [comment]
@srvoin we added a check to switch to --python_path for bazel >=0.5.3 at head, but it was not in 1.3 branch.
https://github.com/tensorflow/tensorflow/commit/6252d296040e8a2ff0c6c8668db55abe3e6f1416#diff-ade1d3e4b7c35655f854151d899df62b 

Not sure if the python_path change is the cause of the issue you are seeing, but might be good to try the configure from master.

**Comment by quaeler** [comment]
I am able to build clean the pip wheel target from head of master (as of 9.30a (pacific) today) via `bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package` - no non-default values in the build configuration setup.

I'm using 0.5.3 Bazel:
```
autogenic:tensorflow loki$ bazel version
Build label: 0.5.3-homebrew
Build target: bazel-out/darwin_x86_64-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Sun Aug 6 12:54:10 2017 (1502024050)
Build timestamp: 1502024050
Build timestamp as int: 1502024050
autogenic:tensorflow loki$

**Comment by martinwicke** [comment]
Thanks @quaeler. I will close this issue, since it will be resolved in 1.4 then.
